src/oslogin_utils.cc

```diff
@@ -428,7 +440,7 @@ bool HttpDo(const string& url, const string& data, string* response, long* http_
         return false;
       }
       curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, http_code);
-    } while (retry_count++ < kMaxRetries && *http_code == 500);
+    } while (retry_count++ < kMaxRetries && ShouldRetry(*http_code));
```

Comment:
Are we idle retrying? Should we add a sleep and a back off strategy?

Reply:
Done